### PR TITLE
Add strategy deployer script

### PR DESCRIPTION
**Pull request by gpsanant**
Add a script for deploying strategies on Goerli

### CODE REVIEW

File: script/misc/DeployStrategy.s.sol [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+import "forge-std/Script.sol";
+import "forge-std/Test.sol";
+
+// # To load the variables in the .env file
+// source .env
+
+// # To deploy and verify our contract
+// forge script script/misc/DeployStrategy.s.sol:DeployStrategy --rpc-url $RPC_URL  --private-key $PRIVATE_KEY --broadcast -vvvv
+
+// NOTE: ONLY WORKS ON GOERLI
```
Comment (ChaoticWalrus) [inline script/misc/DeployStrategy.s.sol:21]: maybe add "Goerli" to the script name then?

File: script/misc/DeployStrategy.s.sol [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    ProxyAdmin public eigenLayerProxyAdmin = ProxyAdmin(0x28ceac2ff82B2E00166e46636e2A4818C29902e2);
+    PauserRegistry public eigenLayerPauserReg = PauserRegistry(0x7cB9c5D6b9702f2f680e4d35cb1fC945D08208F6);
+    StrategyBase public baseStrategyImplementation = StrategyBase(0x9d7eD45EE2E8FC5482fa2428f15C971e6369011d);
+
+    function run() external {
+        // read and log the chainID
+        uint256 chainId = block.chainid;
+        emit log_named_uint("You are deploying on ChainID", chainId);
+
+        address tokenAddress = 0x3ecCAdA3e11c1Cc3e9B5a53176A67cc3ABDD3E46;
```
Comment (ChaoticWalrus) [inline script/misc/DeployStrategy.s.sol:35]: assuming you just change this hard-coded variable for any deployment?

**Review by ChaoticWalrus — APPROVED** [review]
Looks fine to me. Added a couple minor gripes but I'd feel OK with this as-is.

File: script/misc/DeployStrategy.s.sol (hunk 2 above)
Comment (gpsanant) [reply script/misc/DeployStrategy.s.sol:35]: :+1:

File: script/misc/DeployStrategy.s.sol (hunk 1 above)
Comment (gpsanant) [reply script/misc/DeployStrategy.s.sol]: added

**Pull request merged**